### PR TITLE
Refactor `GEN_*` related code

### PR DESCRIPTION
**Pull request by pinkwah**
_see commit descriptions_

### CODE REVIEW

File: src/clib/lib/enkf/gen_data.cpp [hunk 1]
```diff
@@ -200,10 +199,8 @@ void gen_data_deserialize(gen_data_type *gen_data, node_id_type node_id,
         const gen_data_config_type *config = gen_data->config;
```
Comment (oyvindeide) [inline src/clib/lib/enkf/gen_data.cpp:199]: No longer used

File: src/clib/lib/enkf/gen_data.cpp [hunk 2]
```diff
@@ -186,9 +186,8 @@ void gen_data_serialize(const gen_data_type *gen_data, node_id_type node_id,
     const gen_data_config_type *config = gen_data->config;
```
Comment (oyvindeide) [inline src/clib/lib/enkf/gen_data.cpp:186]: No longer used

File: src/clib/lib/enkf/gen_data.cpp [hunk 3]
```diff
@@ -534,37 +484,19 @@ const char *gen_data_get_key(const gen_data_type *gen_data) {
 
 C_USED void gen_data_clear(gen_data_type *gen_data) {
     const gen_data_config_type *config = gen_data->config;
```
Comment (oyvindeide) [inline src/clib/lib/enkf/gen_data.cpp:486]: Not used

**Review by oyvindeide — COMMENTED** [review]
Changes looks good, do you know how the test coverage for this is though? There is probably some reading of ascii files, but binary for GEN_DATA I have not seen.

File: src/clib/lib/enkf/field.cpp [hunk 4]
```diff
@@ -115,65 +114,20 @@ void field_export3D(const field_type *field, void *_target_data,
...
-    default:
-        fprintf(stderr, "%s: Sorry field has unexportable type ... \n",
+
+    const float *src_data = (const float *)field->data;
+    const float *initial_src_data =
+        initial_field ? (const float *)initial_field->data : NULL;
+    if (ecl_type_is_float(target_data_type)) {
+        float *target_data = (float *)_target_data;
+        EXPORT_MACRO;
+    } else if (ecl_type_is_double(target_data_type)) {
```
Comment (ManInFez) [inline src/clib/lib/enkf/field.cpp:124]: is this not always float?

File: src/clib/lib/enkf/field.cpp [hunk 5]
```diff
@@ -833,8 +683,7 @@ static int __get_index(const field_type *field, int i, int j, int k) {
 
 void field_ijk_get(const field_type *field, int i, int j, int k, void *value) {
     int index = __get_index(field, i, j, k);
-    int sizeof_ctype = field_config_get_sizeof_ctype(field->config);
-    memcpy(value, &field->data[index * sizeof_ctype], sizeof_ctype);
+    memcpy(value, &field->data[index * sizeof(float)], sizeof(float));
 }
 
 double field_ijk_get_double(const field_type *field, int i, int j, int k) {
```
Comment (ManInFez) [inline src/clib/lib/enkf/field.cpp:689]: should this also be removed?

File: src/clib/lib/include/ert/enkf/field.hpp [hunk 6]
```diff
@@ -17,7 +17,6 @@
 void field_scale(field_type *field, double scale_factor);
 extern "C" double field_iget_double(const field_type *, int);
 extern "C" double field_ijk_get_double(const field_type *field, int, int, int);
-float field_iget_float(const field_type *, int);
```
Comment (ManInFez) [inline src/clib/lib/include/ert/enkf/field.hpp:20]: I thought that field was only float and not double? :shrug:

Comment (pinkwah) [reply src/clib/lib/include/ert/enkf/field.hpp:20]: Fields contain `float`s but the rest of ERT likes to cast it to `double`, so we only keep the other version of `iget`.

File: src/clib/lib/enkf/field.cpp (hunk 4 above)
Comment (pinkwah) [reply src/clib/lib/enkf/field.cpp:124]: Field data is always `float`. The target data may not be. It's determined by RMS I think.

File: src/clib/lib/enkf/field.cpp (hunk 5 above)
Comment (pinkwah) [reply src/clib/lib/enkf/field.cpp:689]: Yeah the function is unused, removed.

**Comment by ManInFez** [comment]
test this please!

**Comment by pinkwah** [comment]
Jenkins test this please

**Review by ManInFez — APPROVED** [review]

**Pull request merged**